### DISCUSSION TITLE
setting Handled to true doesn't cancel the KeyDown

### DISCUSSION THREAD

**Discussion by Aytharn** [fsprojects/Avalonia.FuncUI#368] [Q&A]
In a Textbox, cancelling the e.Handled by setting it to true (for instance in a textbox I want to ignore non numeric chars) is absolutely useless: yet for some reasons it can cancel Control Keys like cancel and backspace, but no alphanumerics.
Is it a bug?

I even tried to set RoutingStrategies.Direct but nothing changes

**Comment by JaggerJo** [comment]
FuncUI just exposes Avalonia events. It works exactly the same as in Avalonia.
